modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationChanger.java

```diff
@@ -210,36 +193,27 @@ public void register(ConfigurationStorage configurationStorage) throws Configura
             superRoot.addRoot(rootKey, rootNode);
         }
 
-        StorageRoots storageRoots = new StorageRoots(superRoot, data.changeId());
+        storageRoots = new StorageRoots(superRoot, data.changeId());
 
-        storagesRootsMap.put(configurationStorage.type(), storageRoots);
-
-        configurationStorage.registerConfigurationListener(changedEntries -> updateFromListener(
-            configurationStorage.type(),
-            changedEntries
-        ));
+        storage.registerConfigurationListener(this::updateFromListener);
     }
 
     /**
      * Initializes the configuration storage - reads data and sets default values for missing configuration properties.
-     * @param storageType Storage type.
+     *
      * @throws ConfigurationValidationException If configuration validation failed.
-     * @throws ConfigurationChangeException If configuration framework failed to add default values and save them to
-     *      storage.
+     * @throws ConfigurationChangeException If configuration framework failed to add default values and save them to storage.
      */
-    public void initialize(ConfigurationType storageType) throws ConfigurationValidationException, ConfigurationChangeException {
-        ConfigurationStorage configurationStorage = storageInstances.get(storageType);
-
-        assert configurationStorage != null : storageType;
-
+    public void initializeDefaults() throws ConfigurationValidationException, ConfigurationChangeException {
         try {
-            ConfigurationSource defaultsCfgSource = new ConfigurationSource() {
+            ConfigurationSource defaultsCfgSrc = new ConfigurationSource() {
```

Comment:
I believe you're using abbreviation plugin. Can we use "source" as a full word if it's a part of the name?